### PR TITLE
Kickoff-complete

### PR DESCRIPTION
**Pull request by ricardoamoreno**
## Summary

 [Kick-Off](https://trello.com/c/eQB2heIM)

kick-off card finished

## Test Plan

* create postgres database 
* add database credentials on .env
* run project with: npm run start-dev

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = require('eslint-config-wolox-node');
```
Comment (martinremis) [inline .eslintrc.js:1]: Take a look at this line, where ESLint is configured with a set of custom rules. Usually you should add here any change to the slint rules.

Comment (martinremis) [reply .eslintrc.js:1]: (There is nothing to do, just a comment)

File: app.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  limit: config.common.api.bodySizeLimit || DEFAULT_BODY_SIZE_LIMIT
+});
+
+const app = express();
+
+// Client must send "Content-Type: application/json" header
+app.use(bodyParser.json(bodyParserJsonConfig()));
+app.use(bodyParser.urlencoded(bodyParserUrlencodedConfig()));
+app.use(expressRequestIdMiddleware());
+app.use('/docs', swaggerUi.serve, swaggerUi.setup(documentation));
```
Comment (martinremis) [inline app.js:31]: Here there is the Swagger UI setup, which allow you to see in a browser the swagger documentation

File: app/logger/index.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+const { logger } = require('express-wolox-logger');
```
Comment (martinremis) [inline app/logger/index.js:1]: This works with [Pino](https://www.npmjs.com/package/pino) under the hood

File: app/models/index.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+const path = require('path');
+const Sequelize = require('sequelize');
+const config = require('../../config');
+const dbConfig = require('../../config/db')[config.environment];
+
+const basename = path.basename(__filename);
+const db = {};
+const sequelize = new Sequelize(dbConfig.database, dbConfig.username, dbConfig.password, dbConfig);
+
+fs.readdirSync(__dirname)
```
Comment (martinremis) [inline app/models/index.js:11]: This method load the model files to avoid listing them.

File: config/index.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    headers: {
+      apiDate: process.env.API_DATE || 'X-API-Date',
+      packageVersion: process.env.PACKAGE_VERSION || 'X-Package-Version',
+      nodeVersion: process.env.NODE_VERSION || 'X-Node-Version'
+    }
+  }
+};
+
+const customConfig = require(configFile).config;
+module.exports = deepMerge(config, customConfig);
```
Comment (martinremis) [inline config/index.js:52]: This will be the source of all config variables that will be used on the project.

File: migrations/index.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
+const Umzug = require('umzug');
+const config = require('./../config/');
+const { sequelize } = require('../app/models');
+const logger = require('../app/logger');
+
+exports.check = () => {
```
Comment (martinremis) [inline migrations/index.js:6]: This handle the database migrations, loading the files in the directory and comparing if there is a need of running the migration or not, against the database.

File: package.json [hunk 7]
```diff
@@ -0,0 +1,111 @@
+{
+  "name": "WTraining",
+  "version": "0.1.0",
+  "description": "WTraining",
+  "engines": {
+    "node": "10.14.1",
```
Comment (martinremis) [inline package.json:6]: The Node version is deprecated, but in order to follow this training, we will use it anyway.

**Review by martinremis — APPROVED** [review]
I have written some comments, just as information

File: package.json (hunk 7 above)
Comment (ricardoamoreno) [reply package.json:6]: I will update it

**Pull request merged**